src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/Authentication/RemoteAppAuthenticationAuthHandler.cs

```diff
@@ -65,7 +65,14 @@ private async Task<RemoteAppAuthenticationResult> GetRemoteAppAuthenticationResu
 
     protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
     {
-        var authResult = await GetRemoteAppAuthenticationResultAsync();
+        var authResult = await GetRemoteAppAuthenticationResultAsync();
+
+        // Propagate headers back to the caller. Authentication may reissue an authentication cookie, for instance, so the headers need to be returned
+        // to the caller to ensure they are not lost.
+        foreach (var header in authResult.ResponseHeaders)
+        {
+            Context.Response.Headers.Append(header.Key, header.Value);
+        }
```

Comment:
Since this is the same logic as in `HandleChallengeAsync`, it might be better to move it to `GetRemoteAppAuthenticationResultAsync` (just after applying processors), that way the code doesn't need to be duplicated and just runs once right after getting the result.